runelite-client/src/main/java/net/runelite/client/plugins/timetracking/farming/FarmingTracker.java

```diff
@@ -100,13 +104,16 @@ public boolean updateData(WorldPoint location)
 			}
 		}
 
-		for (FarmingRegion region : farmingWorld.getRegions().get(location.getRegionID()))
+		Collection<FarmingRegion> newRegions = farmingWorld.getRegionsForLocation(location);
+
+		if (!newRegions.equals(lastRegions))
 		{
-			if (!region.isInBounds(location))
-			{
-				continue;
-			}
+			newRegionLoaded = true;
+			log.debug("Farming: New region loaded. " + newRegions.toString() + " at " + client.getTickCount() + " ticks");
```

Comment:
Use `{}` replacement rather than string concatenation.
```suggestion
			log.debug("Farming: New region loaded. {} at {} ticks", newRegions, client.getTickCount());
```